File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.py
def solution(k, score):
    top = []
    answer = []
    
    for i, v in enumerate(score):
        if i < k or v > top[0]:
            top.append(v)
            top.sort()
            if len(top) > k:
                top.pop(0)
        answer.append(top[0])
    
    return answer

File: 프로그래머스/1/138477. 명예의 전당 （1）/명예의 전당 （1）.java
import java.util.PriorityQueue;

class Solution {
    public int[] solution(int k, int[] scores) {
        int length = scores.length;
        int[] answer = new int[length];
        PriorityQueue<Integer> top = new PriorityQueue<>();
        
        for (int i=0; i<length; i++) {
            top.add(scores[i]);
            if (top.size() > k) top.poll();
            answer[i] = top.peek();
        }
        
        return answer;
    }
}